omegaconf/omegaconf.py

```diff
@@ -760,13 +763,34 @@ def _node_wrap(
     ref_type: Any = None,
 ) -> Node:
     node: Node
-    is_dict = type(value) is dict or is_dict_annotation(type_)
+    is_dict = (
+        is_dict_annotation(type_)
+        or type_ is dict
+        or (type_ is Any and type(value) is dict)
```

Comment:
I think `type_` here should actually be called ref_type.
See how it's used.

Reply:
I changed it and it fails tests with ref_type